onnxscript/rewriter/pattern.py

```diff
@@ -879,7 +977,8 @@ def replace_pattern(new_pattern):
 
 def _apply_deltas(
     graph_or_function: ir.Graph | ir.Function,
-    deltas: list[tuple[int, tuple[list[ir.Node], list[ir.Node]]]],
+    # TODO(jutinchuby): Use a more descriptive data structure to store deltas
+    deltas: Sequence[tuple[int, tuple[Sequence[ir.Node], Sequence[ir.Node]]]],
```

Comment:
I think we should also be simplifying this function (several aspects). But, as this is a complicated enough PR, I am in favor of not adding more restructuring yet in this PR ...

Reply:
I wonder if in the next PR (combine generic_pattern.py and pattern.py) we would also like to merge function rewriter. function rewriter is currently using protobuf.

Reply:
I would recommend doing those two separately, they are distinct.